controller/aks-cluster-config-handler.go

```
@@ -0,0 +1,686 @@
+				// already in progress. It is possible an update is not being immediately
+				// reflected in upstream cluster state. The config object will reenter the
+				// controller and then the controller will wait for the update to finish.
+				message = err.Error()
+			}
+		}
+
+		if config.Status.FailureMessage == message {
+			return config, err
+		}
+
+		config = config.DeepCopy()
+		if message != "" {
+			if config.Status.Phase == aksConfigActivePhase {
+				// can assume an update is failing
+				config.Status.Phase = aksConfigUpdatingPhase
+			}
+		}
+		config.Status.FailureMessage = message
+
+		var recordErr error
+		config, recordErr = h.aksCC.UpdateStatus(config)
+		if recordErr != nil {
+			logrus.Errorf("Error recording akscc [%s] failure message: %s", config.Name, recordErr.Error())
+		}
+		return config, err
+	}
+}
+
+func (h *Handler) CreateDeployment(ctx context.Context, config *aksv1.AKSClusterConfig) (*aksv1.AKSClusterConfig, error) {
+
+	logrus.Infof("Creating deployment for config [%s]", config.Spec.DisplayName)
+
+	azureAuthorizer, _ := newClientAuthorizer(config.Spec)
+
+	resourceGroupsClient, _ := newResourceGroupsClient(azureAuthorizer, config.Spec)
+	groupExists, _ := checkResourceGroup(ctx, resourceGroupsClient, config.Spec)
+
+	logrus.Infof("Checking if resource group [%s] exists", config.Spec.ResourceGroup)
+
+	if !groupExists {
+		logrus.Infof("Creating resource group [%s] for config [%s]", config.Spec.ResourceGroup, config.Spec.DisplayName)
+		status, err := createResourceGroup(ctx, resourceGroupsClient, config.Spec)
+		if !status || err != nil {
+			logrus.Errorf("Error creating resource group [%s] with message %v", config.Spec.ResourceGroup, err)
+		}
+		logrus.Infof("Resource grup [%s] created successfully", config.Spec.ResourceGroup)
+	}
+
+	logrus.Infof("Creating cluster [%s] for config [%s]", config.Spec.ClusterName, config.Spec.DisplayName)
+	resourceClusterClient, _ := newClustersClient(azureAuthorizer, config.Spec)
```

Are we ignoring this error on purpose?